### PR TITLE
Update InnerBlocks defaultblock doc usage

### PR DESCRIPTION
**Pull request by colorful-tones**
## What?
Recently the `defaultBlock` was stabilized as part of WP 6.4 (https://github.com/WordPress/gutenberg/pull/52083). However, the [`InnerBlocks` `README` docs](https://github.com/WordPress/gutenberg/blob/HEAD/packages/block-editor/src/components/inner-blocks/README.md#defaultblock) were incomplete. I received additional clarification from @sethrubenstein (see here: https://github.com/WordPress/gutenberg/pull/52610#issuecomment-1836279543)

## Why?
Clear instructions for extenders on how to use `defaultBlock`

## How?
Adds clarity to documentation.

I verified while creating a custom block with `useInnerBlocksProps()`. Here is the essential code that worked.

```javascript
const innerBlocksProps = useInnerBlocksProps(
    defaultBlock: {
        name: 'core/cover',
        attributes: {
            dimRatio: 0,
	    layout: {
		type: 'constrained',
	    },
        },
    },
    directInsert: true,
);
```

props: @sethrubenstein

### CODE REVIEW

File: packages/block-editor/src/components/inner-blocks/README.md [hunk 1]
```diff
@@ -189,7 +189,16 @@ For example, a button block, deeply nested in several levels of block `X` that u
 ### `defaultBlock`
 
 - **Type:** `Array`
-- **Default:** - `undefined`. Determines which block type should be inserted by default and any attributes that should be set by default when the block is inserted. Takes an array in the form of `[ blockname, {blockAttributes} ]`.
+- **Default:** - `undefined`. Determines which block type should be inserted by default and any attributes that should be set by default when the block is inserted. Takes an array in the form of `[ name: blockname, attributes: {blockAttributes} ]`.
+
+```jsx
+const DEFAULT_BLOCK = [ name: 'core/paragraph', attributes: { content: 'Lorem ipsum...' } ];
```
Comment (youknowriad) [inline packages/block-editor/src/components/inner-blocks/README.md]: This seems wrong to me, it's either an array or an object, not a mix of both (this notation doesn't work in JavaScript)

Comment (colorful-tones) [reply packages/block-editor/src/components/inner-blocks/README.md]: @youknowriad thanks for catching that! I've updated the accepted type to `Object`.

**Review by youknowriad — APPROVED** [review]

File: packages/block-editor/src/components/inner-blocks/README.md [hunk 2]
```diff
@@ -188,8 +188,17 @@ For example, a button block, deeply nested in several levels of block `X` that u
 
 ### `defaultBlock`
 
-- **Type:** `Array`
-- **Default:** - `undefined`. Determines which block type should be inserted by default and any attributes that should be set by default when the block is inserted. Takes an array in the form of `[ blockname, {blockAttributes} ]`.
+- **Type:** `Object`
+- **Default:** - `undefined`. Determines which block type should be inserted by default and any attributes that should be set by default when the block is inserted. Takes an object in the form of `{ name: blockname, attributes: {blockAttributes} }`.
```
Comment (youknowriad) [inline packages/block-editor/src/components/inner-blocks/README.md]:
```suggestion
- **Default:** - `undefined`

Determines which block type should be inserted by default and any attributes that should be set by default when the block is inserted. Takes an object in the form of `{ name: blockname, attributes: {blockAttributes} }`.
```

It seems weird that the description is in the same line as "default". I moved it after with the above suggestion.

**Comment by youknowriad** [comment]
Thanks for clarifying these docs 👍

**Pull request merged**